frontend/src/pages/singleProduct/singleProduct.js

```diff
@@ -4,19 +4,33 @@ import Gallery from "../../components/gallery/gallery";
 import ProductInfo from "../../components/productInfo/productInfo";
 import productService from "../../services/product.service";
 import { useParams } from "react-router-dom";
-
 import "./singleProduct.css";
+import Notification from "../../components/notification/notification";
+import { STATUS_CODES } from "../../utils/httpStatusCode";
 
 function SingleProduct() {
   const params = useParams();
   const [product, setProduct] = useState();
+  const [showNotification, setShowNotification] = useState(false);
+  const [notificationType, setNotificationType] = useState("");
+  const [notificationMessage, setNotificationMessage] = useState("");
   useEffect(() => {
     productService
       .getSelectedProduct(parseInt(params.productId))
       .then((response) => {
-        setProduct(response.data);
+        if (response.status === STATUS_CODES.OK) setProduct(response.data);
       });
   }, [params]);
+
+  function displayNotification(type, message) {
+    setNotificationType(type);
+    setShowNotification(true);
+    setNotificationMessage(message);
+    setTimeout(function () {
```

Comment:
`Notification` component should be responsible for `setTimeout` like you done before